SMTP_handler.py

```diff
@@ -0,0 +1,520 @@
+from email.message import Message
+from email.utils import formataddr, formatdate
+from init_app import load_pgp_public_keys
+from server import create_light_app
+from app.db import Session
+from app.log import LOG, set_message_id
+from app.email import status, headers
+from app.utils import sanitize_email
+from app.email.spam import get_spam_score
+from app.pgp_utils import PGPException
+from app.models import Alias, SMTPCredentials, EmailLog, Contact, Mailbox
+from app.config import (
+    SMTP_SSL_KEY_FILE,
+    SMTP_SSL_CERT_FILE,
+    NOREPLY,
+    ENABLE_SPAM_ASSASSIN,
+    SPAMASSASSIN_HOST,
+    MAX_REPLY_PHASE_SPAM_SCORE,
+    BOUNCE_PREFIX_FOR_REPLY_PHASE,
+    LOAD_PGP_EMAIL_HANDLER,
+)
+from app.email_utils import (
+    copy,
+    save_email_for_debugging,
+    sanitize_header,
+    get_header_unicode,
+    parse_full_address,
+    is_valid_alias_address_domain,
+    get_spam_info,
+    delete_all_headers_except,
+    sl_sendmail,
+    add_or_replace_header,
+    should_add_dkim_signature,
+    add_dkim_signature,
+    send_email,
+    render,
+    get_email_domain_part,
+    should_ignore_bounce,
+)
+
+from email_handler import (
+    send_no_reply_response,
+    get_or_create_contact,
+    handle_spam,
+    prepare_pgp_message,
+    replace_original_message_id,
+)
+
+
+def handle_SMTP(envelope, msg: Message, rcpt_to: str) -> (bool, str):
+    """
+    Return whether an email has been delivered and
+    the smtp status ("250 Message accepted", "550 Non-existent email address", etc)
+    """
+    website_email = rcpt_to
+
+    # region Create variables For Logging Purpose
+    alias_address: str = envelope.mail_from
+    alias = Alias.get_by(email=alias_address)
+    if not alias:
+        LOG.e("Alias: %s isn't known", alias_address)
+        return False, status.E503
+
+    alias_domain = alias_address[alias_address.find("@") + 1 :]
+
+    # Sanity check: verify alias domain is managed by SimpleLogin
+    # scenario: a user have removed a domain but due to a bug, the aliases are still there
+    if not is_valid_alias_address_domain(alias.email):
+        LOG.e("%s domain isn't known", alias)
+        return False, status.E503
+
+    contact = Contact.get_by(website_email=website_email)
+    if not contact:
+        LOG.w(f"No contact with {website_email} as website email")
+    try:
+        LOG.d("Create or get contact for website_email:%s", website_email)
+        contact = get_or_create_contact("", website_email, alias, msg)
+    except ObjectDeletedError:
+        LOG.d("maybe alias was deleted in the meantime")
+        alias = Alias.get_by(email=alias_address)
+        if not alias:
+            LOG.i("Alias %s was deleted in the meantime", alias_address)
+            if should_ignore_bounce(envelope.mail_from):
+                return [(True, status.E207)]
+            else:
+                return [(False, status.E515)]
+
+    user = alias.user
+    mailbox = Mailbox.get_by(id=alias.mailbox_id)
```

Comment:
Well, if user can have multiple inboxes, it won't be relevant to include mailbox here. 
I added it only for logging as I thought Email Log requires it.  But it doesn't, so it can be removed. What are your thoughts. ?

Reply:
The `mailbox_id` is needed by EmailLog. In this reply phase, this is the mailbox (or a mailbox's authorized address) that sends the email.

Reply:
But for SMTP phase, the `envelope.mail_from`, will be the alias itself.

So how to get the relevant mailbox in case of multiple mailboxes ?

I think it just just be removed as we are already setting `is_SMTP=True`